### PR TITLE
Add the growing spheres generator

### PR DESCRIPTION
**Pull request by kmariuszk**
- Adds the Growing Spheres generator
- Thoroughly tests and documents the functionality

### CODE REVIEW

File: students_documentation/README.md [hunk 1]
```diff
@@ -83,6 +83,12 @@ In this project, we focus on contributing to CounterfactualExplanations.jl, a tr
 
 #### Generators
 
+- Growing Spheres
+  - Core logic: `src/generators/non_gradient_based/growing_spheres/growing_spheres.jl`
+  - Tests: TODO
```
Comment (RaunoArike) [inline students_documentation/README.md:88]: Replace this with the correct path

File: test/runtests.jl [hunk 2]
```diff
@@ -34,23 +34,23 @@ include("utils.jl")
...
     end
 
     @testset "Probe" begin
         include("generators/probe.jl")
     end
 
     @testset "CLUE" begin
-        include("clue.jl")
+        include("generators/clue.jl")
     end
```
Comment (RaunoArike) [inline test/runtests.jl:54]: CLUE and PROBE are already included in generators.jl and don't need to be included here.

**Review by RaunoArike — CHANGES_REQUESTED** [review]
Left two comments, otherwise seems really solid!

File: test/runtests.jl (hunk 2 above)
Comment (kmariuszk) [reply test/runtests.jl]: Done

**Review by RaunoArike — APPROVED** [review]
LGTM

**Review by VincentPikand — APPROVED** [review]
clean code

**Pull request merged**